src/Controller/SubjectsController.php

```diff
@@ -61,7 +64,14 @@ public function subjectAction(Request $request, string $id) : Response
 
         $item = $this->get('elife.api_sdk.subjects')
             ->get($id)
-            ->otherwise($this->mightNotExist());
+            ->otherwise($this->mightNotExist())
+            ->then(function (Subject $subject) use ($id) {
+                if ($subject->getId() !== $id) {
+                    throw new EarlyResponse(new RedirectResponse($this->get('router')->generate('subject', [$subject])));
```

Comment:
Might be something to look at doing generally - the API could be saying that something has moved.

Reply:
As in `301 Moved Permanently`, I guess

Reply:
Yeah. Not actually sure if it's configured to follow redirects. Probably should...